internal/app/secretless/handlers/mysql/backend.go

```diff
@@ -82,31 +89,80 @@ func (h *Handler) ConnectToBackend() (err error) {
 		return
 	}
 
+	requestedSSL := true
+
+	// requested SSL but server doesn't support it
+	if requestedSSL && serverHandshake.ServerCapabilities&protocol.ClientSSL == 0 {
+		return &protocol.Error{
+			Code:     protocol.CRSSLConnectionError,
+			SQLSTATE: protocol.ErrorCodeInternalError,
+			Message:  ErrNoTLS.Error(),
+			SequenceID: 2,
+		}
+	}
+
 	// Intercept response from client
 	handshakeResponsePacket, err := protocol.ReadPacket(h.GetClientConnection())
 	if err != nil {
 		return
 	}
+	// Track clientSequenceID
+	clientSequenceID := handshakeResponsePacket[3]
 
 	// Parse intercepted client response
 	handshakeResponse, err := protocol.UnpackHandshakeResponse41(handshakeResponsePacket)
 	if err != nil {
 		return
 	}
 
+	// Ensure CapabilityFlag is set when using TLS
+	if requestedSSL {
+		handshakeResponse.CapabilityFlags |= protocol.ClientSSL
+	}
+
 	// Inject credentials into client response
 	if err = protocol.InjectCredentials(handshakeResponse, serverHandshake.Salt, h.BackendConfig.Username, h.BackendConfig.Password); err != nil {
 		return
 	}
 
 	// Pack client response with injected configuration
-	clientPacket, err := protocol.PackHandshakeResponse41(handshakeResponse)
+	packedHandshakeRespPacket, err := protocol.PackHandshakeResponse41(handshakeResponse)
+
 	if err != nil {
 		return
 	}
 
+	if requestedSSL {
+		// Send TLS / SSL request packet
+
+		// Copy a truncated HandshakeResponse to create SSLRequest
+		// https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::SSLRequest
+		tmp := packedHandshakeRespPacket[:(4+4+1+23)+4]
+		sslPacket := make([]byte, len(tmp))
+		copy(sslPacket, tmp)
+
+		// Update packet length for truncated packet
+		pktLen := len(sslPacket) - 4
+		sslPacket[0] = byte(pktLen)
+		sslPacket[1] = byte(pktLen >> 8)
+		sslPacket[2] = byte(pktLen >> 16)
+
```

Comment:
This stuff took my a long time to read and I had to go the links, and I still don't understand it all.  I think it should include more extensive comments.    In particular I'm confused  how the capability flags aren't being overwritten here.